oak_channel/src/frame.rs

```diff
@@ -117,9 +118,11 @@ impl Framed {
             let body_length: usize = length
                 .checked_sub(BODY_OFFSET)
                 .expect("body length underflow");
-            let body = allocate_fn(body_length);
-            self.inner.read(body)?;
-            body
+            let tail = message_buffer.len();
+            // Lack of capacity indicates corrupted frames and causes panic.
+            message_buffer.put_bytes(0x00, body_length);
+            self.inner.read(&mut message_buffer[tail..])?;
+            message_buffer
```

Comment:
Huh, I'm somewhat surprised by this (but I guess the tests pass so who am I to say..) -- I'd expected this to be `&message_buffer[tail..]` or something similar here.

Reply:
Thanks for catching this. We need testing to catch this.